hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/ResourceHistoryCalculatorTest.java

```diff
@@ -0,0 +1,318 @@
+			Arguments.of(FhirContext.forR4Cached(), false, ResourceEncodingEnum.ESR, EXCLUDED_ELEMENTS_2),
+			Arguments.of(FhirContext.forR4Cached(), true, ResourceEncodingEnum.JSON, EXCLUDED_ELEMENTS_2),
+			Arguments.of(FhirContext.forR4Cached(), false, ResourceEncodingEnum.JSON, EXCLUDED_ELEMENTS_1)
+		);
+	}
+
+	@ParameterizedTest
+	@MethodSource("calculateResourceHistoryStateArguments")
+	void calculateResourceHistoryState(FhirContext theFhirContext, boolean theIsOracle, ResourceEncodingEnum theResourceEncoding, List<String> theExcludedElements) {
+		final IBaseResource patient = getPatient(theFhirContext);
+
+		final ResourceHistoryCalculator calculator = getCalculator(theFhirContext, theIsOracle);
+		final ResourceHistoryState result = calculator.calculateResourceHistoryState(patient, theResourceEncoding, theExcludedElements);
+
+		if (theIsOracle) {
+			assertNotNull(result.getResourceBinary());
+			assertNull(result.getResourceText());
+			assertEquals(theResourceEncoding, result.getEncoding());
+			assertEquals(SHA_256.hashBytes(result.getResourceBinary()), result.getHashCode());
+		} else {
+			assertNull(result.getResourceBinary());
+			assertNotNull(result.getResourceText());
+			assertEquals(ResourceEncodingEnum.JSON, result.getEncoding());
+			final HashCode expectedHashCode = SHA_256.hashUnencodedChars(calculator.encodeResource(patient, theResourceEncoding, theExcludedElements));
+			assertEquals(expectedHashCode, result.getHashCode());
+		}
+	}
+
+
+	private static Stream<Arguments> conditionallyAlterHistoryEntityArguments() {
+		return Stream.of(
+			Arguments.of(true, ResourceEncodingEnum.JSONC, ENCODED_RESOURCE_1),
+			Arguments.of(true, ResourceEncodingEnum.JSONC, ENCODED_RESOURCE_2),
+			Arguments.of(true, ResourceEncodingEnum.DEL, ENCODED_RESOURCE_1),
+			Arguments.of(true, ResourceEncodingEnum.DEL, ENCODED_RESOURCE_2),
+			Arguments.of(true, ResourceEncodingEnum.ESR, ENCODED_RESOURCE_1),
+			Arguments.of(true, ResourceEncodingEnum.ESR, ENCODED_RESOURCE_2),
+			Arguments.of(true, ResourceEncodingEnum.JSON, ENCODED_RESOURCE_1),
+			Arguments.of(true, ResourceEncodingEnum.JSON, ENCODED_RESOURCE_2),
+			Arguments.of(false, ResourceEncodingEnum.JSONC, ENCODED_RESOURCE_1),
+			Arguments.of(false, ResourceEncodingEnum.JSONC, ENCODED_RESOURCE_2),
+			Arguments.of(false, ResourceEncodingEnum.DEL, ENCODED_RESOURCE_1),
+			Arguments.of(false, ResourceEncodingEnum.DEL, ENCODED_RESOURCE_2),
+			Arguments.of(false, ResourceEncodingEnum.ESR, ENCODED_RESOURCE_1),
+			Arguments.of(false, ResourceEncodingEnum.ESR, ENCODED_RESOURCE_2),
+			Arguments.of(false, ResourceEncodingEnum.JSON, ENCODED_RESOURCE_1),
+			Arguments.of(false, ResourceEncodingEnum.JSON, ENCODED_RESOURCE_2)
+		);
+	}
+
+	@ParameterizedTest
+	@MethodSource("conditionallyAlterHistoryEntityArguments")
+	void conditionallyAlterHistoryEntity(boolean theIsOracle, ResourceEncodingEnum theResourceEncoding, String theResourceText) {
```

Comment:
minor: this name doesn't help me understand what the test is going to do.
```suggestion
	void conditionallyAlterHistoryEntity_usesVarcharForOracle(boolean theIsOracle, ResourceEncodingEnum theResourceEncoding, String theResourceText) {
```